lib/jobs/ansible-job.js

```diff
@@ -78,10 +95,23 @@ function ansibleJobFactory(
      * @memberOf AnsibleJob
      */
      AnsibleJob.prototype._cleanup = function cleanup() {
+        var self = this;
+        try {
+            var data = fs.readFileSync(self.outFileName, 'utf8');
```

Comment:
Either way, it would probably be better to use a promisified version of the async readFile call, that way the job can't complete and pass execution to the next task (which may read context) before it is actually done reading the data and setting the context. You can see some example usage of this here:

Promisify fs methods: https://github.com/RackHD/on-core/blob/0be650eb0c5d21f4ee854463ab0133d3d878fea6/lib/common/file-loader.js#L17
Call the promisified method: https://github.com/RackHD/on-core/blob/0be650eb0c5d21f4ee854463ab0133d3d878fea6/lib/common/file-loader.js#L48